lib/inets/src/http_server/httpd.erl

```diff
@@ -45,16 +45,75 @@
          info/4
         ]).
 
+-export_type([ httpd_information/0
+             , httpd_properties/0
+             , httpd_comm_properties/0
+             , httpd_mods_properties/0
+             , httpd_limit_properties/0
+             , httpd_admin_properties/0]).
+
 -deprecated({parse_query, 1,
             "use uri_string:dissect_query/1 instead"}).
 
 %%%========================================================================
 %%% API
 %%%========================================================================
 
+-type httpd_information() :: [httpd_properties()]
+                           | [httpd_comm_properties()]
+                           | [httpd_mods_properties()]
+                           | [httpd_limit_properties()]
+                           | [httpd_admin_properties()]
+                           | [{atom(), term()}].
+
+-type httpd_properties() :: {port, non_neg_integer()}
+                          | {server_name, string()}
+                          | {server_root, path()}
+                          | {document_root, path()}.
+
+-type path() :: file:name_all().
+
+-type httpd_comm_properties() ::
+                            {bind_address, inet:ip_address() | inet:hostname() | any}
+                          | {profile, atom()}
+                          | { socket_type, ip_comm | {ip_comm, proplists:proplist()} | {essl, proplists:proplist()}}
```

Comment:
I think this is the place where @KennethL  would like the documentation to say ssl instead of essl (that will have the same effect ). It also should say [ssl:ssl_option]  and I think [gen_tcp:option()]  instead of proplists:proplist. To make this change of essl -> ssl I think we ought to change the code in http_transport also to make essl the wrapper and ssl the default as not it is actually the other way around even if that is not what we think is best.

Reply:
Got it. I'll see if I can add some more types around. Thanks